app/lib/string_encryptor.rb

```diff
@@ -0,0 +1,36 @@
+class StringEncryptor
+  KEY_LEN = ActiveSupport::MessageEncryptor.key_len
+
+  def initialize(signing_key:)
+    @signing_key = signing_key
+  end
+
+  def encrypt_string(plaintext)
+    salt = SecureRandom.hex KEY_LEN
+    ciphertext = encryptor(salt).encrypt_and_sign(plaintext)
+    Base64.urlsafe_encode64("#{salt}$$#{ciphertext}")
+  end
+
+  def decrypt_string(ciphertext)
+    bits = Base64.urlsafe_decode64(ciphertext).split("$$")
+    return nil unless bits.length == 2
+
+    salt = bits[0]
+    message = bits[1]
+
+    encryptor(salt).decrypt_and_verify(message)
+  rescue ArgumentError
+    nil
+  rescue ActiveSupport::MessageEncryptor::InvalidMessage
+    nil
```

Comment:
Do we need to explicitly return `nil` here, or is it for readability?

Reply:
I thought it was more readable, but I'm happy to change it if you think not

Reply:
I asked because I was wondering if after a `rescue` the default return value wasn't `nil` (as it usually is) and we needed it explicitly. Even if it is, I think there's value in being explicit after rescuing the errors.